### ISSUE TITLE
Trouble opening ports

### ISSUE THREAD

**Issue by Wahhhhhhhh** [pi-node/instructions#113]
I've tried all of the instructions on https://node-pi.com/pinode_eng but still no luck.
Don't know where to even start, any help would be appreciated.
![image](https://user-images.githubusercontent.com/40413473/116961009-f99e5400-ac56-11eb-9f66-8135c7e83b5d.png)

Don't know if it's relevant but there is no default gateway displayed under vEthernet when I type ipconfig in CMD.

**Comment by ihatejam** [comment]
What were your port test results?

**Comment by Wahhhhhhhh** [comment]
> What were your port test results?

It says NOT OPEN for all of the ports
![image](https://user-images.githubusercontent.com/40413473/117052964-37d65a80-accd-11eb-985d-00dff6eb5693.png)

**Comment by steeveee100** [comment]
First thing is you need to upgrade Win10 to 19042

**Comment by Wahhhhhhhh** [comment]
It is on that version currently

**Comment by nathusius** [comment]
@Wahhhhhhhh advice from other recent threads: the built-in portscan check is temperamental. You can sanity-check your ports at https://www.ipfingerprints.com/portscan.php: with the stellar-dummy running in Docker, you should see all ports open here (you will need to type in the port number range 31400 to 31409)
For me the built-in port checker never even attempted to test anything, but ipfingerprints worked fine with all ten ports showing as open and I ended up getting picked up OK.
Also, did you need to setup port forwarding on your router? If so, re-check that this is setup correctly, as I found mine got messed up after a reboot and I needed to allocate a fixed IP address to my network adapter, so that the port forwarding as still valid after a PC reboot (this is for wired connection to a router)

**Comment by Wahhhhhhhh** [comment]
@nathusius Thanks for the advice.
So with what you said, I've gone and checked the ports on ipfingerprints.
they all say "filtered" which means a firewall or AV is filtering?
I've tried disabling my AV and as far as the firewall, I've open the ports as the guide says :
![image](https://user-images.githubusercontent.com/40413473/117247492-2edfa900-adf3-11eb-9d3b-512bb188c684.png)

And yes, I've set up the port forwarding on my router 
![image](https://user-images.githubusercontent.com/40413473/117247718-a6153d00-adf3-11eb-86cb-9691d9853b87.png)
The internal IP is set up correctly on my screen, I just erased the digits just to be safe?

I really can't figure out what is filtering the ports, it's so frustrating...

**Comment by ihatejam** [comment]
@Wahhhhhhhh If you've set up port forwarding but ipfingerprints still reports that your ports are closed/filtered then there's one more check to do -- 
Write down the public IP address ipfingerprints thinks you're on [Note: do NOT post your public IP address here].
Compare the public IP address reported by ipfingerprints with the public IP address reported by the WAN interface on your router (you'll need to log into your router config -- and it is NOT the address starting with 192. ). If the two IP addresses do not match, then it is likely that your ISP does not hand out "real" public IP addresses. This is common practice in places where the demand for IP addresses is larger than the IP address block allocated to your ISP. To resolve this, you can do the following:
a) ask your ISP if they can give you an IP address that allows port forwarding (warning: their support guys may have no clue what you mean and/or they may not be able to honor your request).
b) switch to an ISP that allows/facilitates port forwarding.
c) use a VPN provider that automatically takes care of port forwarding like [Surfshark](https://surfshark.club/friend/b5BmtEtn?locale=en)

**Comment by nathusius** [comment]
@Wahhhhhhhh Yes, it seems that you have done everything possible on the firewall & router, so you will need to follow advice just above from @ihatejam - sorry, I do know how frustrating this is!

**Comment by Wahhhhhhhh** [comment]
@nathusius Thank you for helping!

@ihatejam Just check that again to make sure, my IP on both fingerprints and the WAN are the same unfortunately... Anything else I could check?

**Comment by pilgrim103** [comment]
A drastic thing would be to buy your own router and not use the crap you get with cable companies.  I am almost to that point but got mine to work.

**Comment by ihatejam** [comment]
@Wahhhhhhhh If ipfingerprints says the ports are filtered and your IP addresses match up, then you would need to contact your ISP about allowing you to forward those ports. Chances are that they currently only allow the default ports through for their subscribers - like ports needed for web-browsing/gaming/etc.

If you have no joy, [Surfshark VPN]( https://surfshark.club/friend/b5BmtEtn) is your friend - running it can get around all of this config issues for you. Bbear in mind that there is currently no reward for running a Pi node before you decide to pay for a VPN service.

**Comment by Wahhhhhhhh** [comment]
@pilgrim103 as a matter of fact I do run my own device and not the ISPs :/

**Comment by Wahhhhhhhh** [comment]
@ihatejam yeah I think I'm going to wait until there is a definite gain from running the node to start using the VPN, thanks for all your help!!

**Comment by ihatejam** [comment]
@Wahhhhhhhh does the troubleshooting page still show you the portchecker is running and consensus=error??

If not, please post an updated screenshot of your troubleshooting screen.

**Comment by ihatejam** [comment]
Circling back to your first post... (With explanation)

You can only have one container running at a time - when the consensus container starts up, the portchecker container is meant to auto-stop. 

In the first screenshot, it appears as if you manually started the portchecker, which would have prevented the consensus container from starting up. Going through the motions of checking the ports and then flipping the node switch on (to red) should have resolved it.

**Comment by pilgrim103** [comment]
You have the wrong person.  I never posted a photo.

I also have to Nodes (wife and I actually) running the blockchain.

Cheers
________________________________

**Comment by pilgrim103** [comment]
You have the wrong person.  I never posted a photo.

I also have to Nodes (wife and I actually) running the blockchain.

Cheers
________________________________

**Comment by ihatejam** [comment]
@pilgrim103 i think you've misread my previous responses  - unless I've @'d you, they're not aimed at you but at the at the owner this thread - ( Lois Chung aka Wahhhhhhhh )

**Comment by pilgrim103** [comment]
thanks
________________________________

**Comment by brokenisuseless** [comment]
I had a lot of these same issues, until I went into the NIC properties and changed the send and receive buffers to the max of 2048, I believe default is 512 or 1024. Current setup debian linux running windows8.1 in virtualbox, running pi node there using Docker Toolbox.. the virtualbox running windows is using a bridged adapter, so it gets an IP right from my router, I forwarded the ports to that IP,then I had to forward the ports in the windows virtualbox that is running docker. some hardware firewalls in the past have had issues with double nat type setups , that was years ago but it could still happen. PS all of this had huge issues for me, while using IP6 I turned that off and it started working

**Comment by pilgrim103** [comment]
Sounds like Pi needs to get all this port stuff straightened out before they go any further.  I have never seen such a confusing mess.  I am sooo tired of hearing about it.  Is anyone at Stanford listening or are they already counting their money?  The Port Checker they tell you to use rarely works.  If it does it lies.  Some people are using Linux solutions, other Virtual Box NAT solutions, some Docker (which is a hackers dream come true) but most others are giving up.  All this when we have been told not to expect any rewards for being a SuperNode/Node/Blockchaining.  Everyday 5 new coins get released and some of them actually have brilliant ideas and whitepapers.  Has anyone read the whitepaper for Pi? Social issues?  I thought it was great at first when I heard they were planning on using cell phones to build and release a new Cryptocurrency.  The cell app cannot even be called that, it does nothing.  It is not even correct.  Check the new one with the 18+ Million message (and they think they will be getting a 100 million?).  It has a label of "now" at the top showing that the current coin reward is 0.2 pi/hr.  Check your phone.  It is 0.1 + whatever incentives you have earned from invites or Security Circles.  Cannot even get that right.  They do have along way to go and need funding BADLY.  Why do you think they play commercials?  Some of the new coins being developed and released are announcing hundreds of millions of dollars in financing and backing.  One of Pi's men quit last year.  There is alot more to point out but when you leave Node installation to non-technical people from third world countries it shows the mindset.  They are determined to release this year and so be it.  I will be there but I will just buy 7,000,000,000 Pi coins for 0.0000010 u.s. dollars like Shiba or Swap Moon.  There are going on 10,000 Cryptocurrencies now.  Is Pi's model better or more effecient than them?  I do not see it.  If you are relying on the Social issue to give them an edge forget it.  It is all about $$$$.  I hope they succeed because I will be buying but this Port issue has become such a joke SOMEONE at Stanford should have been able to straighten this out by now.  Now I will add more "good" news to the Port issue.  I have had varying success with it until I restarted my Xfinity Gateway.  I lost my port forwarding.  Re-instated it and no luck.  Ended up talking to a head Technical engineer at Xfinity and he told me that Pi's plans will not work on any Xfinity router because they have a "different" viewpoint on NAT, they use it differently and he suggested if you have an Xfinity modem to get a separate router.  He did say that some very old models of theirs will work but they now are DESIGNED not to allow what Pi is trying to do.  (Just a heads up on the dreaded Port issue.)  Then their is the Wallet issue, and the "incentive" issue for blockchaining (electricity is expensive where I live, and the windmill in my backyard will not light a light bulb.)  Now who has trouble opening ports?
